RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.h

```diff
@@ -0,0 +1,194 @@
+  const edm::ESGetToken<GlobalTrackingGeometry, GlobalTrackingGeometryRecord> t_geometryH_;
+  const edm::ESGetToken<NavigationSchool, NavigationSchoolRecord> t_navSchool_;
+  const edm::ESGetToken<Propagator, TrackingComponentsRecord> t_SHPOpposite_;
+
+  /// Maximum number of seeds for each L2
+  const unsigned int maxSeeds_;
+
+  /// Maximum number of hitbased seeds for each L2
+  const unsigned int maxHitSeeds_;
+
+  /// Maximum number of hitless seeds for each L2
+  const unsigned int maxHitlessSeeds_;
+
+  /// How many layers to try
+  const unsigned int numOfLayersToTry_;
+
+  /// How many hits to try per layer
+  const unsigned int numOfHitsToTry_;
+
+  /// Rescale L2 parameter uncertainties (fixed error vs pT, eta)
+  const double fixedErrorRescalingForHitless_;
+
+  /// Estimator used to find dets and TrajectoryMeasurements
+  const std::string estimatorName_;
+
+  /// Minimum eta value to activate searching in the TEC
+  const double minEtaForTEC_;
+
+  /// Maximum eta value to activate searching in the TOB
+  const double maxEtaForTOB_;
+
+  /// KFUpdator defined in constructor
+  const std::unique_ptr<TrajectoryStateUpdator> updator_;
+
+  const edm::EDGetTokenT<MeasurementTrackerEvent> measurementTrackerTag_;
+
+  /// Counters and flags for the implementation
+  const std::string propagatorName_;
+  const std::string theCategory_;
+
+  /// IP refers to TSOS at interaction point,
+  /// MuS refers to TSOS at muon system
+  const unsigned int maxHitlessSeedsIP_;
+  const unsigned int maxHitlessSeedsMuS_;
+  const unsigned int maxHitDoubletSeeds_;
+
+  /// Get number of seeds to use from DNN output instead of "max..Seeds" parameters
+  const bool getStrategyFromDNN_;
+
+  /// Whether to use DNN regressor (if false, will use classifier)
+  const bool useRegressor_;
+
+  /// Settings for classifier
+  const double etaSplitForDnn_;
+  std::string dnnModelPath_barrel_;
+  std::string dnnModelPath_endcap_;
+
+  tensorflow::GraphDef* graphDef_barrel_;
+  tensorflow::Session* tf_session_barrel_;
+
+  tensorflow::GraphDef* graphDef_endcap_;
+  tensorflow::Session* tf_session_endcap_;
+
+  /// Settings for regressor
+  std::string dnnModelPath_HB_;
+  std::string dnnModelPath_HLIP_;
+  std::string dnnModelPath_HLMuS_;
+
+  tensorflow::GraphDef* graphDef_HB_;
+  tensorflow::Session* tf_session_HB_;
+
+  tensorflow::GraphDef* graphDef_HLIP_;
+  tensorflow::Session* tf_session_HLIP_;
+
+  tensorflow::GraphDef* graphDef_HLMuS_;
+  tensorflow::Session* tf_session_HLMuS_;
+
+  /// DNN metadata
+  const std::string dnnMetadataPath_;
+  pt::ptree metadata;
+
+  /// Create seeds without hits on a given layer (TOB or TEC)
+  void makeSeedsWithoutHits(const GeometricSearchDet& layer,
+                            const TrajectoryStateOnSurface& tsos,
+                            const Propagator& propagatorAlong,
+                            edm::ESHandle<Chi2MeasurementEstimatorBase>& estimator,
```

Comment:
unless required, please dereference the handles in the caller and pass just the `Chi2MeasurementEstimatorBase&` in the member method calls (same for the other methods)

Reply:
Done, except for the `NavigationSchool` handle in `TSGForOIDNN::makeSeedsFromHitDoublets`.
Let me know if it should be changed as well.

Reply:
Done!